state/model.go

```diff
@@ -871,6 +871,39 @@ func (m *Model) AllUnits() ([]*Unit, error) {
 	return units, nil
 }
 
+type ApplicationEndpointBindings struct {
+	AppName  string
+	Bindings map[string]string
+}
+
+// AllBindings return all endpoint - space bindings deployed.
+func (m *Model) AllBindings() ([]ApplicationEndpointBindings, error) {
+	endpointBindings, closer := m.st.db().GetCollection(endpointBindingsC)
+	defer closer()
+
+	var docs []endpointBindingsDoc
+	err := endpointBindings.FindId(nil).All(&docs)
+	if err != nil {
+		return nil, errors.Annotatef(err, "cannot get endpoint bindings")
+	}
+
+	var appEndpointBindings []ApplicationEndpointBindings
+	for _, doc := range docs {
+		var ApplicationName string
+		// for each service(application service) deployed we have an instance of ApplicationEndpointBindings struct
+		if strings.HasPrefix(doc.DocID, "a#") == true {
```

Comment:
no need for == true
plus, if the prefix doesn't match we should return an error like
errors.NotValidf("application key %v", doc.DocID)